### PR TITLE
Remove Razor Pages specific things from the MVC (Model-View-Controller) template #10935

### PR DESCRIPTION
**Pull request by isaacrlevin**
Summary of the changes (Less than 80 chars)
 - Remove RazorPages extension if template is not an Auth one

Addresses #10935 

Per @rynowak the RazorPages extensions are needed for auth projects because of the Identity library. If the project generated is not an auth one, the extensions are not needed.

### CODE REVIEW

**Comment by isaac2004** [comment]
@ryanbrandenburg flaky test failure

**Comment by ryanbrandenburg** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=238057

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2489

**Review by ryanbrandenburg — APPROVED** [review]
Seems good! I'll merge this once the bot triage's the helix failure.

**Comment by ryanbrandenburg** [comment]
Whoops, I'm still catching up on emails, it appears that we're mostly not merging things into master at the moment. But the PR looks good, we'll merge it for Preview8.

File: src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs [hunk 1]
```diff
@@ -120,7 +120,9 @@ public void ConfigureServices(IServiceCollection services)
 #else
             services.AddControllersWithViews();
 #endif
-            services.AddRazorPages();
+#if (OrganizationalAuth || IndividualAuth)
```
Comment (danroth27) [inline src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs]: Does org auth use razor pages?

Comment (isaac2004) [reply src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs]: I was pretty sure it uses the identity UI. Let me check

Comment (isaac2004) [reply src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Startup.cs]: Confirmed Microsoft.AspNetCore.Authentication.AzureAD.UI uses Razorpages

**Review by danroth27 — APPROVED** [review]

**Pull request merged**